library/Ivoz/Provider/Domain/Model/Voicemail/Voicemail.php

```diff
@@ -34,6 +34,16 @@ public function getId(): ?int
         return $this->id;
     }
 
+    protected function sanitizeValues(): void
+    {
+        $user = $this->getUser();
+        $residentialDevice = $this->getResidentialDevice();
+
+        if (!$user && !$residentialDevice) {
+            throw new \DomainException('User or residential device required');
+        }
```

Comment:
Not totally true.

When voicemail have no user nor residential Device, its called generic voicemail. Only vPBX clients can have generic voicemails.

Voicemails with userId are auto-created from Users. Voicemails with residentialDeviceId are auto-created from Residential Devices. Voicemails without user or residentialDevice can be created and deleted by vPBX  administrators.

Reply:
:+1: